backend/hitas/views/housing_company.py

```diff
@@ -0,0 +1,329 @@
+
+    def get_area(self, obj: HousingCompany) -> Dict[str, Any]:
+        return {"name": obj.postal_code.description, "cost_area": obj.area}
+
+    def get_date(self, obj: HousingCompany) -> datetime.date:
+        return obj.date
+
+    class Meta:
+        model = HousingCompany
+        fields = ["id", "name", "state", "address", "area", "date"]
+
+
+class HousingCompanyDetailSerializer(EnumSupportSerializerMixin, serializers.ModelSerializer):
+    id = serializers.SerializerMethodField()
+    name = serializers.SerializerMethodField()
+    state = serializers.SerializerMethodField()
+    address = serializers.SerializerMethodField()
+    area = serializers.SerializerMethodField()
+    date = serializers.SerializerMethodField()
+    real_estates = serializers.SerializerMethodField()
+    building_type = serializers.SerializerMethodField()
+    financing_method = serializers.SerializerMethodField()
+    developer = serializers.SerializerMethodField()
+    property_manager = serializers.SerializerMethodField()
+    acquisition_price = serializers.SerializerMethodField()
+    primary_loan = serializers.SerializerMethodField()
+    last_modified = serializers.SerializerMethodField()
+    notes = serializers.SerializerMethodField()
+    legacy_id = serializers.SerializerMethodField()
+
+    def get_id(self, obj: HousingCompany) -> str:
+        return obj.uuid.hex
+
+    def get_name(self, obj: HousingCompany) -> Dict[str, str]:
+        return {
+            "display": obj.display_name,
+            "official": obj.official_name,
+        }
+
+    def get_legacy_id(self, obj: HousingCompany) -> Optional[str]:
+        return value_or_none(obj.legacy_id)
+
+    def get_notes(self, obj: HousingCompany) -> Optional[str]:
+        return value_or_none(obj.notes)
+
+    def get_state(self, obj: HousingCompany) -> str:
+        return obj.state.name.lower()
+
+    def get_address(self, obj: HousingCompany) -> Address:
+        return address_obj(obj)
+
+    def get_area(self, obj: HousingCompany) -> Dict[str, any]:
+        return {"name": obj.postal_code.description, "cost_area": obj.area}
+
+    def get_date(self, obj: HousingCompany) -> datetime.date:
+        return obj.date
+
+    def get_building_type(self, obj: HousingCompany) -> Dict[str, Any]:
+        return BuildingTypeSerializer(obj.building_type).data
+
+    def get_financing_method(self, obj: HousingCompany) -> Dict[str, Any]:
+        return FinancingMethodSerializer(obj.financing_method).data
+
+    def get_developer(self, obj: HousingCompany) -> Dict[str, Any]:
+        return DeveloperSerializer(obj.developer).data
+
+    def get_property_manager(self, obj) -> Dict[str, Any]:
+        return PropertyManagerSerializer(obj.property_manager).data
+
+    def get_acquisition_price(self, obj: HousingCompany) -> Dict[str, float]:
+        return {
+            "initial": to_float(obj.acquisition_price),
+            "realized": to_float(obj.realized_acquisition_price),
+        }
```

Comment:
Why are we manually casting decimals to floats? 🤔

Reply:
ah, `to_float` is useless. But without using `SerializerMethodField` we get this error 
```
<ValidationError: "'10.00' is not of type number">
<ValidationError: "'10.00' is not a 'float'">
```

`COERCE_DECIMAL_TO_STRING=False` fixes this (or `DecimalField(coerce_to_string=False)`)